src/pinns_pde_solve.jl

```diff
@@ -282,80 +293,49 @@ function generate_training_sets(domains,dx,bcs,indvars,depvars,dict_indvars,dict
     spans = [d.domain.lower:dx:d.domain.upper for (d,dx) in zip(domains,dxs)]
     dict_var_span = Dict([Symbol(d.variables) => d.domain.lower:dx:d.domain.upper for (d,dx) in zip(domains,dxs)])
 
-    train_set = []
-    for points in Iterators.product(spans...)
-        push!(train_set, [points...])
-    end
+    train_set = map(points -> [points...], Iterators.product(spans...))
 
-    train_bound_set = []
-    for bt in bound_args
-        _set = []
-        span = [get(dict_var_span, b, b) for b in bt]
-        for points in Iterators.product(span...)
-            push!(_set, [points...])
-        end
-        push!(train_bound_set, _set)
+    train_bound_set = map(bound_args) do bt
+        span = map(b -> get(dict_var_span, b, [b]), bt)
+        _set = map(points -> [points...], Iterators.product(span...))
     end
 
-    flat_train_bound_set = collect(Iterators.flatten(train_bound_set))
+    flat_train_bound_set = map(x -> x, Iterators.flatten(train_bound_set))
     train_domain_set =  setdiff(train_set, flat_train_bound_set)
 
     [train_domain_set,train_bound_set,train_set]
 end
 
-function get_phi(chain,isuinplace)
+function get_phi(chain)
+    chain isa Chain && "Chain deprecated, use FastChain"
+    isuinplace = chain.layers[end].out == 1
     # The phi trial solution
-    if chain isa FastChain
-        if isuinplace
-            phi = (x,θ) -> first(chain(adapt(typeof(θ),x),θ))
-        else
-            phi = (x,θ) -> chain(adapt(typeof(θ),x),θ)
-        end
+    if isuinplace
+        phi = (x,θ) -> first(chain(adapt(typeof(θ),x),θ))
     else
-        _,re  = Flux.destructure(chain)
-        if isuinplace
-            phi = (x,θ) -> first(re(θ)(adapt(typeof(θ),x)))
```

Comment:
do you need to destructure here?

Reply:
no

Reply:
oh actually I think you do if you're going to use sciml_train over Flux Chain here.

Reply:
I confused with the Flux.destructor from PhysicsInformedNN

Reply:
yeah, it needs here